main.tf

```diff
@@ -0,0 +1,362 @@
+    #filter_suffix = ".json"
+  }
+}
+
+#create CSV bucket notification
+resource "aws_s3_bucket_notification" "csv_bucket_trigger_sns" {
+  #bucket = aws_s3_bucket.csv-bucket.id
+  bucket = var.csv_bucket_name
+
+  topic {
+    topic_arn     = aws_sns_topic.conversion_complete_topic.arn
+    events        = ["s3:ObjectCreated:*"]
+    #filter_suffix = ".csv"
+  }
+}
+
+##########################################################################################
+####################################   CLOUDWATCH  #######################################
+##########################################################################################
+#cloudwatch group creation
+resource "aws_cloudwatch_log_group" "json-csv-log-group" {
+  name              = "/aws/lambda/${var.lambda_function_name}"
+  retention_in_days = 90
+}
+
+##########################################################################################
+####################################   LAMBDA ############################################
+##########################################################################################
+data "aws_iam_policy_document" "assume_role" {
+  statement {
+    effect = "Allow"
+
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+
+    actions = ["sts:AssumeRole"]
+  }
+}
+
+#create an iam role for lambda
+resource "aws_iam_role" "iam_for_lambda" {
+  name               = "iam_for_lambda"
+  assume_role_policy = data.aws_iam_policy_document.assume_role.json
+}
+
+#SQS permissions for lambda
+data "aws_iam_policy_document" "allow_lambda_to_receiveSQSMessage" {
+  statement {
+    effect    = "Allow"
+    actions   = [
+      "sqs:ReceiveMessage",
+      "sqs:DeleteMessage",
+      "sqs:GetQueueAttributes"
+    ]
+    resources = [aws_sqs_queue.JSON_event_queue.arn]
+
+    condition {
+      test     = "ArnEquals"
+      variable = "aws:SourceArn"
+      values   = [aws_lambda_function.csv_to_json_lambda.arn]
+    }
+  }
+}
+
+resource "aws_iam_policy" "lambda_SQS_recieve" {
+  name        = "lambda_SQS_recieve_name"
+  path        = "/"
+  description = "IAM policy for logging from a lambda"
+  policy      = data.aws_iam_policy_document.allow_lambda_to_receiveSQSMessage.json
+}
+
+resource "aws_iam_role_policy_attachment" "lambda_SQS" {
+  role       = aws_iam_role.iam_for_lambda.name
+  policy_arn = aws_iam_policy.lambda_SQS_recieve.arn
+}
+
+#CloudWatch permissions for lambda
+data "aws_iam_policy_document" "lambda_logging" {
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "logs:CreateLogGroup",
+      "logs:CreateLogStream",
+      "logs:PutLogEvents",
+    ]
+    resources = ["${aws_cloudwatch_log_group.json-csv-log-group.arn}:*"]
+  }
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "sqs:ReceiveMessage",
+      "sqs:DeleteMessage",
+      "sqs:GetQueueAttributes"
+    ]
+
+    resources = [aws_sqs_queue.JSON_event_queue.arn]
+  }
```

Comment:
Redundant permissions, these actions are already allowed in `allow_lambda_to_receiveSQSMessage` Line 172